sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -603,6 +603,13 @@ object SQLConf {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefault(10L)
 
+  val STREAMING_NO_DATA_REPORT_INTERVAL =
+    SQLConfigBuilder("spark.sql.streaming.noDataReportInterval")
+      .internal()
+      .doc("How long to wait between two progress events when there is no data")
```

Comment:
TimeConf supports various units like `10s`, `100ms`. The unit here is telling which unit we want to use when getting this conf.